### PR TITLE
Enable more ruff rules

### PR DESCRIPTION
**Pull request by jelly**
Also sneaked in a unit test improvement as waiting for 6 seconds got annoying ;-)

### CODE REVIEW

File: task/test-cache [hunk 1]
```diff
@@ -55,26 +55,26 @@ class TestCache(unittest.TestCase):
         self.assertEqual(result, other)
 
     def testCurrent(self):
-        c = cache.Cache(self.directory, lag=3)
+        c = cache.Cache(self.directory, lag=0.5)
```
Comment (allisonkarlitskaya) [inline task/test-cache:58]: I don't like this test, and I particularly don't like your changes to it... this timing-dependent stuff is asking for trouble, and making the window smaller makes it more dangerous.

Why not introduce a better separation between linting and unit tests?

**Review by martinpitt — APPROVED** [review]
LGTM, but please finish the discussion with Lis first. Dankjewel!

File: task/test-cache (hunk 1 above)
Comment (jelly) [reply task/test-cache:58]: I ran the test ~ 10 times in a row and that seemed fine, this is mostly a pre-push hook annoyance as that takes 6 seconds. Hmm can we have a different approach of testing this.

Comment (jelly) [reply task/test-cache]: Dropped it, as I don't feel like bikeshedding in this PR, we can discuss it separate from this PR :)

**Review by martinpitt — APPROVED** [review]

**Review by allisonkarlitskaya — APPROVED** [review]
Fine by me too, but the aliasing fix looks like you were hoping to `select = [ "A" ]` as well?

**Comment by jelly** [comment]
> Fine by me too, but the aliasing fix looks like you were hoping to `select = [ "A" ]` as well?

Yup, but that requires more possibly API breaking changes, as some public API uses builtin's

**Pull request merged**